test/commands/force/data/bulk/status.test.ts

```diff
@@ -4,136 +4,196 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import { $$, expect, test } from '@salesforce/command/lib/test';
-import { stubMethod } from '@salesforce/ts-sinon';
-import { Org } from '@salesforce/core';
+/* eslint-disable @typescript-eslint/no-explicit-any */
 
-interface StatusResult {
-  status: number;
-  message: string;
-}
+import * as path from 'path';
+import { TestContext, MockTestOrgData } from '@salesforce/core/lib/testSetup';
+import { Config } from '@oclif/core';
+import { expect } from 'chai';
+import Bulk, { Batch, BulkOperation, Job, JobInfo, JobState } from 'jsforce/lib/api/bulk';
+import stripAnsi = require('strip-ansi');
+import Status from '../../../../../src/commands/force/data/bulk/status';
 
 describe('force:data:bulk:status', () => {
-  test
-    .do(() => {
-      const Job = {
-        getAuthInfoFields: () => {
-          return { orgId: '123' };
-        },
-        bulk: {
-          job: () => {
-            return {
-              id: '75054000006yv68AAA',
-              state: 'Open',
-              options: {},
-              type: null,
-              operation: null,
-              list: () => {
-                return [
-                  {
-                    id: '751540000070aNsAAI',
-                    jobId: '75054000006yv68AAA',
-                    state: 'Completed',
-                    createdDate: '2020-11-30T16:14:21.000Z',
-                    systemModstamp: '2020-11-30T16:14:21.000Z',
-                    numberRecordsProcessed: '1',
-                    numberRecordsFailed: '1',
-                    totalProcessingTime: '185',
-                    apiActiveProcessingTime: '55',
-                    apexProcessingTime: '0',
-                  },
-                ];
-              },
-            };
-          },
-        },
-      };
-      stubMethod($$.SANDBOX, Org.prototype, 'getConnection').returns(Job);
-    })
-    .withOrg({ username: 'test@org.com' }, true)
-    .stdout()
-    .command([
-      'force:data:bulk:status',
-      '--targetusername',
-      'test@org.com',
-      '--batchid',
-      '751540000070Q4RAAU',
-      '--jobid',
-      '75054000006ybyHAAQ',
-      '--json',
-    ])
-    .it('will fail with the correct error message', (ctx) => {
-      const result = JSON.parse(ctx.stdout) as StatusResult;
-      expect(result.message).to.equal('Unable to find batch 751540000070Q4RAAU for job 75054000006ybyHAAQ.');
-      expect(result.status).to.equal(1);
-    });
+  const $$ = new TestContext();
```

Comment:
If you do `TestContext({setup: true})`, you don't need to worry instantiating or restoring the defaults stubs: https://github.com/forcedotcom/sfdx-core/blob/main/src/testSetup.ts#L364

Reply:
ooh, we should doc that over in sfdx-core's readme or on one of the /examples/test/unit